vllm/entrypoints/openai/api_server.py

```diff
@@ -129,7 +129,8 @@ async def check_length(
         input_ids = tokenizer(prompt).input_ids
     token_num = len(input_ids)
 
-    if token_num + request.max_tokens > max_model_len:
+    unlimited_tokens = request.max_tokens == max_model_len
+    if not unlimited_tokens and token_num + request.max_tokens > max_model_len:
```

Comment:
Why does this `if` needs to be changed?

Reply:
If `None` is directly passed to `request.max_tokens`, this if statement will complains something like `TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'`. So I set `request.max_tokens=max_model_len` to avoid this type mismatch.

And after `request.max_tokens` is set to `max_model_len`, `token_num + max_model_len` will always be larger than `max_model_len`. So I add a `unlimited_tokens` condition to try to avoid this. 

In fact, this change is just for the special case that we pass `max_tokens=None` in the request for a unlimited number of tokens during generation.

Reply:
And actually this code may not be able to differ the situation mentioned above from the case we pass `max_tokens=max_model_len` in the request. Maybe there is a better way, like set a large enough num to represent the `inf` and reset it to the max available num?

Reply:
Can't you just set `request.max_tokens` to `max_model_len - token_num` if `request.max_tokens` is `None` so that you only allow, at maximum, to generate the number of tokens remaining before filling the context size?

```py
@@ async def check_length
   ... 

   if request.max_tokens is None:
        request.max_tokens = max_model_len - token_num
    elif token_num + request.max_tokens > max_model_len:
        return input_ids, create_error_response(
            HTTPStatus.BAD_REQUEST,
            f"This model's maximum context length is {max_model_len} tokens. "
            f"However, you requested {request.max_tokens + token_num} tokens "
            f"({token_num} in the messages, "
            f"{request.max_tokens} in the completion). "
            f"Please reduce the length of the messages or completion.",
        )
    
    return input_ids, None
```

Reply:
Thank you for the reminder. This solution is better. I have revised my code.